ammo-fe/src/network/MockedSocket.ts

```diff
@@ -1,14 +1,29 @@
+// Arguments can be either a callback function (in the case of the toggleRecord for example), or a list of unknowned arguments
+type Arguments = unknown[] | ((recording: boolean) => void);
+
 export default class MockedSocket {
     private cache: Record<string, (...args: unknown[]) => void> = {};
 
+    private isRecording = true;
+
     public connected = true;
 
     on = (event: string, listener: (...args: unknown[]) => void): void => {
         this.cache[event] = listener;
     };
 
-    emit = (event: string, ...args: unknown[]): void => {
-        this.cache[event](...args);
+    emit = (event: string, args: Arguments): void => {
+        // When we toggle, we save the new value and invoke the callback given just like the server would do
+        if (event === 'toggleRecord') {
+            this.isRecording = !this.isRecording;
+            if (typeof args === 'function') {
+                args(this.isRecording);
+            }
+            return undefined;
+        }
+        // For now, we only block the bullet event when the recorder is paused
+        if (!this.isRecording && event === 'bullet') return undefined;
+        return this.cache[event](args);
```

Comment:
I don't get why we have to be this specific for the emit() method, and have a this.isRecording boolean (when there's already a state in Redux for that). Let's talk about it